dhall/src/Dhall/Main.hs

```diff
@@ -159,6 +159,23 @@ data ResolveMode
     | ListTransitiveDependencies
     | ListImmediateDependencies
 
+-- | Groups of subcommands
+data Group
+  = Manipulate
+  | Generate
+  | Interpret
+  | Convert
+  | Miscellaneous
+  | Debugging
+
+groupDescription :: Group -> String
+groupDescription group = case group of
+  Manipulate -> "Manipulate Dhall code"
+  Generate -> "Generate other formats from Dhall"
+  Interpret -> "Interpret Dhall"
+  Convert -> "Convert Dhall to and from its binary representation"
+  Miscellaneous -> "Miscellaneous"
+  Debugging -> "Debugging this interpreter"
```

Comment:
I would slightly prefer the new code to use the same indentation with 4 spaces as the surrounding code, but it's not very important. :)

Reply:
Oh, sorry about that. One sec.

Reply:
How's that?